lib/runner/variant-reassignment.js

```diff
@@ -71,17 +73,31 @@ export default class VariantReassignment {
     return Promise.resolve()
   }
 
-  async _handleProcessingError (productDraft, products) {
+  /**
+   * For 400 errors, we don't repeat the actions as the request itself is wrong.
+   * For other errors, we retry the actions.
+   *
+   * @see https://github.com/commercetools/commercetools-node-variant-reassignment/issues/60
+   */
+  async _handleProcessingError (productDraft, products, error) {
+    if (error.statusCode === 400) {
+      this.logger.info('Unrecoverable error, will delete backup custom object.')
```

Comment:
Should we delete transaction from the custom object? What if there are some variants which were not created yet? Maybe we can just move them somewhere so we can later analyze what went wrong?

Reply:
You can supply an error callback. There you can do whatever you want. Neverthless, I created a task for this, maybe we can do it in the future: https://github.com/commercetools/commercetools-node-variant-reassignment/issues/62